js/mbankScrapeRegular_onefile.js

```diff
@@ -0,0 +1,389 @@
+  }
+
+  toArray() {
+    return [
+      this.date_col,
+      this.memo_col.replace(/,/g, " "),
+      this.payee_col,
+      "-" + this.amount_col.slice(0, -4).replace(/,/g, ".")
+    ];
+  }
+}
+class Prowizja extends Transakcja {
+  get date_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Data rozliczenia')]]/td").innerHTML;
+  }
+
+  toArray() {
+    return [
+      this.date_col,
+      "",
+      this.memo_col.replace(/,/g, " "),
+      this.amount_col.slice(0, -4).replace(/,/g, ".")
+    ];
+  }
+}
+class Mtransfer extends Transakcja {
+  get date_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Data księgowania')]]/td").innerHTML;
+  }
+  get payee_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Nazwa odbiorcy')]]/td").innerHTML;
+  }
+  get memo_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Tytuł przelewu')]]/td").innerHTML;
+  }
+
+  toArray() {
+    return [
+      this.date_col,
+      this.memo_col.replace(/,/g, " "),
+      this.payee_col.replace(/,/g, " "),
+      "-" + this.amount_col.slice(0, -4).replace(/,/g, ".")
+    ];
+  }
+}
+class Blik extends Transakcja {
+  get payee_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Nazwa sklepu internetowego')]]/td").innerHTML;
+  }
+  get memo_col() {
+    return "Blik: " + this.getElementByXpath("//tr[th[contains(text(), 'Nr operacji BLIK')]]/td").innerHTML;
+  }
+}
+class PrzelewWychodzacy extends Transakcja {
+  get date_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Data księgowania')]]/td").innerHTML;
+  }
+  get payee_col() {
+    let rachunek_odbiorcy = this.getElementByXpath("//tr[th[contains(text(), 'Nazwa odbiorcy')]]/td").innerHTML;
+    if (rachunek_odbiorcy == lokaty) {
+      return "Transfer to: Lokaty";
+    }
+    return rachunek_odbiorcy;
+  }
+  get memo_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Tytuł przelewu')]]/td").innerHTML;
+  }
+  get rachunek_odbiorcy() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Rachunek odbiorcy')]]/td").innerHTML;
+  }
+
+  toArray() {
+    return [
+      this.date_col,
+      this.memo_col.replace(/,/g, " "),
+      this.payee_col.replace(/,/g, " "),
+      "-" + this.amount_col.slice(0, -4).replace(/,/g, ".")
+    ];
+  }
+}
+class PrzelewPrzychodzacy extends Transakcja {
+  get payee_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Nazwa nadawcy')]]/td").innerHTML;
+  }
+  get memo_col() {
+    return this.getElementByXpath("//tr[th[contains(text(), 'Tytuł przelewu')]]/td").innerHTML;
+  }
+  get nadawca() {
+    let rachunek_nadawcy = this.getElementByXpath("//tr[th[contains(text(), 'Rachunek nadawcy')]]/td").innerHTML;
+    if (! [emax, ekonto, emaxplus].includes(rachunek_nadawcy) ) {
+      return rachunek_nadawcy;
+    }
+    return console.log("Ignored incoming from own account: " + this.amount_col + " (" + this.date_col + ")");
```

Comment:
In previous version this check was done like this:
```
    else if (MEMO_COL.includes("PRZYCHODZĄCY")) {
      // ignore incoming from own accounts
      if (! [emax, ekonto, emaxplus].includes(incoming.nadawca) ) {
        arr.push(incoming.toArray());
      } else {
        console.log("Ignored incoming from own account: " + incoming.amount_col + " (" + incoming.date_col + ")");
      }
    }
```
If I have this here, I won't be able to make the saveDetailsForCSV method to skip this transaction. How to handle this?